packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -90,17 +90,17 @@ export const usePostTypeMenuItems = ( onClickMenuItem ) => {
 	const publicPostTypes = usePublicPostTypes();
 	const existingTemplates = useExistingTemplates();
 	const defaultTemplateTypes = useDefaultTemplateTypes();
-	// `page`is a special case in template hierarchy.
-	const templatePrefixes = useMemo(
+
+	const templateEntityObject = useMemo(
 		() =>
-			publicPostTypes?.reduce( ( accumulator, { slug } ) => {
-				let suffix = slug;
-				if ( slug !== 'page' ) {
-					suffix = `single-${ suffix }`;
-				}
-				accumulator[ slug ] = suffix;
-				return accumulator;
-			}, {} ),
+			publicPostTypes?.map( ( { slug } ) => {
```

Comment:
Can we keep the structure we had before(object and `reduce`)? I don't see much value here for this change and my rationale is:
1. we only need something extra for `user` and we can just pass an extra param with the extra query args in `useEntitiesInfo`
2. I don't really like the assignment with `index` in the loop below.. It's very indirect and in case we'll need to find something outside of looping the entities, we'll need to loop again with `find` or something.

Reply:
I refactored this and added an additional parameter.